stable/sumologic-fluentd/templates/secrets.yaml

```diff
@@ -8,9 +8,6 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
-  annotations:
```

Comment:
Will this cause any issues for users who already have the chart installed with the hooks registered?

Reply:
We are using helmfile, and when I try to upgrade the chart I get this error

```[us-1-west.mertslounge.ca] sumologic> helmfile sync
Upgrading sumologic-fluentd-0.12.1.tgz
Error: UPGRADE FAILED: no Secret with the name "sumologic-sumologic-fluentd" found

err: release "sumologic" in "helmfile.yaml" failed: exit status 1
exit status 1```

Reply:
Something (helmfile?) is overriding `sumologic-fluentd.fullname` that defaults to the chart name ("sumologic-fluentd"), not "sumologic-sumologic-fluentd" like your error is showing. Check what is the real name of the secret you already have.

Reply:
That is what the helper function generates:

https://github.com/helm/charts/blob/1ba310f11226dae48d07245e5f03d992eab8f8c5/stable/sumologic-fluentd/templates/_helpers.tpl#L14-L15

Here is the same output when using helm directly

```
> helm upgrade sumologic ./sumologic-fluentd-0.12.1.tgz --values values.yaml
Error: UPGRADE FAILED: no Secret with the name "sumologic-sumologic-fluentd" found
```

I have not experienced an issue upgrading to any other chart version, it only happens when the hooks are removed from the secret. I suspect its an issue with removing hooks in helm

https://github.com/helm/helm/issues/3785#issuecomment-400659576

Reply:
Im not sure how to proceed with this change. While I agree with the direction, the upgrade implications could be significant. The typical helm way to deal with this is likely to delete and reinstall the chart with the new version.  However, this could be problematic for customers who make use of `readFromHead` if they do not have a good persistence strategy.

We could ask the helm community if there another way to manage removal of the hook

@frankreno thoughts?

Reply:
Correct me if I'm wrong, but this is no different than any other upgrade circumstance, right? Pretty sure a host path volume will not be cleaned up upon pod deletion, since otherwise they would be useless for preserving state between pods when rolling the daemonset. If you delete the existing release with persistence enabled, a new release should pick up the position files on each node. 

Without persistence enabled, fluentd is still configured to write position files to the container. They will remain in the event of a restart of the existing container. In those circumstances, _any_ change to the fluentd pod configuration, e.g. almost any chart version , upgrade, adding custom conf, etc. results in a full read from head on all log files. 

I've been thinking about additional documentation or potentially changing the defaults around read-on-head/persistence. I don't think the current defaults are reasonable for use in production and none of the repos involved are adequately up front about:

* How different settings make tradeoffs w.r.t. data loss Sumo rate limiting and how to configure to maximize reliability
* The relationship of this problem/fluentd with log rotation, or lack thereof, in a cluster

I do think persistence may be disabled by default because it relies on settings `type: DirectoryOrCreate` added in k8s@1.8 and persistence has been relatively unchanged in this chart since release. A breaking release would be an ideal opportunity to change that.

Not sure we need to do anything different here for this particular change because of persistence concerns, just the need for manual upgrade. 

The would mean:

* Breaking semver release
* Providing manual upgrade instructions
* Ideally creating/linking to the notes on configuring for reliability I described above

Users running without persistence would be subject to the usual caveats about re-transmission of existing logs to Sumo and associated cluster resource/Sumo rate limit cost.

Reply:
And actually I'm not sure a manual upgrade is even required here. Sounds like your issue may have to do with helmfile, but if you can repro in minikube with plain `helm` that would indicate otherwise.

> Check what is the real name of the secret you already have.

This would still be helpful to know, sounds like there might just be an issue we could fix here if Helm is looking up the wrong name for deletion.

Reply:
Late the party.  So one option is to make the change and do a major version bump on the chart.  Then we can document upgrade instructions on the GitHub readme.  Still, I do not like that as much as people may miss it and then experience the issue. If we do this we can also make default value changes as they have been that way primarily for backward compatibility of older k8s versions.

Reply:
@jessesuen does ^ sound good?